cmd/csi-cluster-driver-registrar/k8s_register.go

```diff
@@ -42,8 +43,9 @@ func kubernetesRegister(
 	}
 
 	// Set up goroutine to cleanup (aka deregister) on termination.
+	// Kubernetes uses SIGTERM, not SIGINT.
 	c := make(chan os.Signal, 1)
-	signal.Notify(c, os.Interrupt)
+	signal.Notify(c, syscall.SIGTERM)
```

Comment:
Why not `signal.Notify(c, os.Interrupt, syscall.SIGTERM)` That way if on windows or unix-flavor operating systems it still works.

Reply:
might be worth adding a comment here on  the reason why we settled on either `SIGTERM` or `Interrupt` or both

Reply:
Okay, will do.

Is there interest in resurrecting the lost code that puts both adding and removal into the main goroutine?

Reply:
I found the code again (different repo, different branch...). I think it's cleaner, therefore I have added it to this PR together with some other minor cleanups.